presto-spi/src/main/java/io/prestosql/spi/block/LongArrayBlockEncoding.java

```diff
@@ -53,9 +65,20 @@ public Block readBlock(BlockEncodingSerde blockEncodingSerde, SliceInput sliceIn
         boolean[] valueIsNull = decodeNullBits(sliceInput, positionCount).orElse(null);
 
         long[] values = new long[positionCount];
-        for (int position = 0; position < positionCount; position++) {
-            if (valueIsNull == null || !valueIsNull[position]) {
-                values[position] = sliceInput.readLong();
+        if (valueIsNull == null) {
+            sliceInput.readBytes(Slices.wrappedLongArray(values));
+        }
+        else {
+            int nonNullPositionCount = sliceInput.readInt();
+            long[] valuesWithoutNull = new long[nonNullPositionCount + 1];
```

Comment:
This extra allocation is going to be a bit of a bummer if you only have a few nulls since it effectively doubles the allocations involved with decoding.

Reply:
this is where we could use buffer cache potentially. Still, even then there is > 2x improvement here (and more then that for serialization)

Reply:
Changed to
```
            long[] valuesWithoutNull = new long[positionCount];
            int nonNullPositionCount = 0;
            for (int i = 0; i < positionCount; i++) {
                valuesWithoutNull[nonNullPositionCount] = block.getLong(i, 0);
                if (!block.isNull(i)) {
                    nonNullPositionCount++;
                }
            }
```
since it yields slightly better results without extra allocation